modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -161,6 +118,8 @@ public TimeValue getCompareAndExchangeTimeToLive() {
     // issue
     private abstract static class IgnoreNoResponseMetricsCollector extends RequestMetricCollector {
 
+        protected final AtomicLong counter = new AtomicLong();
```

Comment:
Seems it might be incremented concurrently. Should we consider using LongAdder here?

Reply:
I was trying to maintain the existing pattern of using AtomicLong. But you are right that LongAdder is more speed performant. Updated.